torch_xla/csrc/xla_backend_impl.cpp

```diff
@@ -0,0 +1,171 @@
+    XLA_ERROR() << "Not implemented yet";
+  }
+
+  const torch::lazy::IrBuilder* GetIrBuilder() const override {
+    XLA_ERROR() << "Not implemented yet";
+    return 0;
+  }
+
+  torch::lazy::BackendDataPtr MakeComputationDataFromTensor(
+      const at::Tensor& tensor, const torch::lazy::Shape& shape,
+      const torch::lazy::BackendDevice& device) const override {
+    return TensorToXlaData(tensor, device, /*transfer_async=*/false);
+  }
+
+  torch::lazy::BackendDataPtr MakeComputationDataFromScalar(
+      const at::Scalar& scalar,
+      const torch::lazy::BackendDevice& device) const override {
+    at::Tensor t = at::scalar_tensor(scalar);
+    return TensorToXlaData(t, device, /*transfer_async=*/false);
+  }
+
+  torch::lazy::BackendDataPtr CreateDataPlaceholder(
+      const torch::lazy::BackendDevice& device,
+      const torch::lazy::Shape& shape) const override {
+    xla::Shape xla_shape = MakeXlaShapeFromLazyShape(shape, device);
+    return WrapXlaData(xla::ComputationClient::Get()->CreateDataPlaceholder(
+        device.toString(), std::move(xla_shape)));
+  }
+
+  torch::lazy::BackendDataPtr GetComputationDataFromNode(
+      torch::lazy::Node* node) const override {
+    const DeviceData* device_data_node = DeviceData::Cast(node);
+    if (!device_data_node) {
+      return nullptr;
+    }
+    return device_data_node->data();
+  }
+
+  at::Tensor MakeTensorFromComputationData(
+      const torch::lazy::BackendDataPtr data,
+      c10::optional<at::ScalarType> logical_scalar_type) const override {
+    // TODO(JackCaoG): handle the logical_scalar_type == nullptr case
+    return XlaDataToTensors({data}, *logical_scalar_type)[0];
+  }
+
+  std::unique_ptr<torch::lazy::LoweringContext> CreateLoweringContext(
+      const std::string& name, torch::lazy::BackendDevice device,
+      c10::ArrayRef<torch::lazy::Node*> post_order,
+      torch::lazy::Util::EmissionMap emit_status) const override {
+    // TODO(JackCaoG): change LoweringContext to take post_order as
+    // c10::ArrayRef<torch::lazy::Node*> instead of
+    // c10::ArrayRef<const torch::lazy::Node*> since c10::ArrayRef already
+    // provided const for its member.
+    return std::make_unique<LoweringContext>(name, device);
+  }
+
+  std::unique_ptr<torch::lazy::LoweringContext> CreateLoweringContext(
+      const std::string& name,
+      torch::lazy::BackendDevice device) const override {
+    return std::make_unique<LoweringContext>(name, device);
+  }
+
+  // GetCompilationDevices always returns devices today because
+  // 1. devices passed to GetCompilationDevices is always empty
+  // 2. ComputationClient's replication device is never set since we always have
+  // one replica per process
+  // We should revisit this API for SPMD
+  std::vector<std::string> GetCompilationDevices(
+      const std::string& device,
+      c10::ArrayRef<std::string> devices) const override {
+    return xla::ComputationClient::Get()->GetCompilationDevices(device,
```

Comment:
fyi @yeounoh, we currently call it in https://github.com/pytorch/xla/blob/185dd3cd45e4a1b9dcbbf88a3c604ce8c5e752f3/torch_xla/csrc/tensor.cpp#L1666.